csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs

```diff
@@ -605,6 +686,82 @@ public static OrtValue CreateTensorValueFromMemory<T>(T[] data, long[] shape) wh
             return OrtValue.CreateTensorValueFromMemory(OrtMemoryInfo.DefaultInstance, new Memory<T>(data), shape);
         }
 
+#if NET8_0_OR_GREATER
+#pragma warning disable SYSLIB5001 // System.Numerics.Tensors is only in preview so it can continue receiving API feedback
+        /// <summary>
+        /// This is a factory method creates a native Onnxruntime OrtValue containing a tensor.
+        /// The method will attempt to pin managed memory so no copying occurs when data is passed down
+        /// to native code.
+        /// </summary>
+        /// <param name="value">Tensor object</param>
+        /// <param name="elementType">discovered tensor element type</param>
+        /// <returns>And instance of OrtValue constructed on top of the object</returns>
+        public static OrtValue CreateTensorValueFromDotnetTensorObject<T>(DotnetTensors.Tensor<T> tensor) where T : unmanaged
```

Comment:
Is it possible to avoid creating a new function and re-use CreateTensorValueFromMemory()? Does the new Tensor exposes Memory<T> property?

Reply:
The new tensor does not expose a Memory property. Let me note that down though as feedback for us and bring it up with my team.